### PR TITLE
Add JS API for modifying the client-side error collection

### PR DESCRIPTION
**Pull request by exyi**
The new API has 2 new functions:
* `dotvvm.validation.addErrors([{ errorMessage: "Test", propertyPath: "/X/Y/Z" }])`
* `dotvvm.validation.removeErrors("/FirstName", "/LastName")`

### CODE REVIEW

File: src/Framework/Framework/Resources/Scripts/validation/validation.ts [hunk 1]
```diff
@@ -280,24 +288,44 @@ function getValidationErrors<T>(
...
 
+export type ValidationErrorDescriptor = {
+    /** Error to be displayed to the user */
+    errorMessage: string
+    /** Path in the view model to the annotated property, for example `/LoginPage/Name` */
+    propertyPath: string
+}
+
+export type AddErrorsOptions = {
+    /** Root object from which are the property paths are resolved. By default it's the root view model of the page */
```
Comment (acizmarik) [inline src/Framework/Framework/Resources/Scripts/validation/validation.ts:299]:
```suggestion
    /** Root object from which are the property paths resolved. By default it's the root view model of the page */
```

File: src/Framework/Framework/Resources/Scripts/validation/validation.ts [hunk 2]
```diff
@@ -280,24 +288,44 @@ function getValidationErrors<T>(
...
+    root?: KnockoutObservable<any> | any
+    /** When set to false, the validationErrorsChanged is not trigger. By default, the error is triggered. The event controls the `Validator`s in the DotHTML page, so when it's not trigger, the change won't be visible. */
```
Comment (acizmarik) [inline src/Framework/Framework/Resources/Scripts/validation/validation.ts:301]: I am confused by this comment. To me, it seems like the `triggerErrorsChanged` is set to `false` by default. Also when is it desirable to add errors but not propagate these changes to `Validator`s?

```suggestion
    /** When set to false, the validationErrorsChanged would not trigger. The event controls the `Validator`s state in the DotHTML page, so when it's not triggered, the change won't be visible. */
```

Comment (exyi) [reply src/Framework/Framework/Resources/Scripts/validation/validation.ts:301]: > Also when is it desirable to add errors but not propagate these changes to Validators?

it's "needed" when the method is called multiple times and you don't want to trigger the event for each invocation. Maybe it's just more confusing, I'll consider removing this (at least for now), it's probably just confusing.

Comment (exyi) [reply src/Framework/Framework/Resources/Scripts/validation/validation.ts]: Actually, we need it for our own usage in the `showValidationErrorsFromServer` function bellow, I couldn't deduplicate the code without this switch.

**Review by acizmarik — APPROVED** [review]
👍

**Pull request merged**